File: Excel/570 Challenge.py
import pandas as pd

path = "570 Position Swapping.xlsx"
input = pd.read_excel(path,usecols="A:B", nrows=10)
test = pd.read_excel(path,usecols="C", nrows=10)

def swap_string_tidy(string, numbers):
    str_list = list(string)
    indices = [int(num) - 1 for num in numbers.split(',')]
    for i in range(0, len(indices), 2):
        str_list[indices[i]], str_list[indices[i+1]] = str_list[indices[i+1]], str_list[indices[i]]
    return ''.join(str_list)

input['Swapped'] = input.apply(lambda row: swap_string_tidy(row['String'], row['Numbers']), axis=1)

print(input['Swapped'].equals(test['Answer Expected'])) # True

File: Excel/500-599/570/570 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/570 Position Swapping.xlsx"
input = read_excel(path, range = "A1:B10")
test  = read_excel(path, range = "C1:C10")

swap_string = function(string, numbers) {
  numbers = as.numeric(str_split(numbers, ",\\s*")[[1]])
  str_vec = str_split(string, "", simplify = TRUE)
  for (i in seq(1, length(numbers), by = 2)) {
    str_vec[c(numbers[i], numbers[i + 1])] = str_vec[c(numbers[i + 1], numbers[i])]
  }
  str_c(str_vec, collapse = "")
}

result = input %>%
  mutate(Swapped = map2_chr(String, Numbers, swap_string))

all.equal(result$Swapped, test$`Answer Expected`)
# [1] TRUE